oshmem/mca/scoll/basic/scoll_basic_alltoall.c

```diff
@@ -135,9 +135,10 @@ static int a2as_alg_simple(struct oshmem_group_t *group,
 
     dst_blk_idx = oshmem_proc_group_find_id(group, group->my_pe);
 
-    for (src_blk_idx = 0; src_blk_idx < group->proc_count; src_blk_idx++) {
+    for (i = 0; i < group->proc_count; i++) {
 
-        dst_pe = get_dst_pe(group, src_blk_idx, dst_blk_idx);
+        dst_pe = get_dst_pe(group, i, dst_blk_idx);
```

Comment:
I think it is not correct now. I suggest modifying get_dst_pe so that you get back (dst_pe, dst_pe_index) where
dst_pe_index = (dst_blk_idx + src_blk_idx) % group->proc_count;
dst_pe           = oshmem_proc_pe(group->proc_array[dst_pe_index])